Orange/widgets/visualize/owscatterplot.py

```diff
@@ -380,6 +387,19 @@ def update_graph(self, attributes=None, reset_view=True, **_):
         self.graph.zoomStack = []
         if attributes and len(attributes) == 2:
             self.attr_x, self.attr_y = attributes
+        data_x = self.data[:, self.attr_x]
+        data_y = self.data[:, self.attr_y]
+        if self.attr_x not in self.attr_nans:
+            self.attr_nans[self.attr_x] = BasicStats(data_x, self.attr_x).nans
+        if self.attr_y not in self.attr_nans:
+            self.attr_nans[self.attr_y] = BasicStats(data_y, self.attr_y).nans
+        self.information()
+        self.warning()
+        if (self.attr_nans[self.attr_x] == len(data_x) or
+            self.attr_nans[self.attr_y] == len(data_y)):
+            self.warning("Some attributes are empty")
```

Comment:
It can be only `attr_x` or `attr_y` (or both) that is empty. That's not a lot of if statements?